atlasdb-jepsen-tests/src/test/java/com/palantir/atlasdb/jepsen/NemesisResilienceCheckerTest.java

```diff
@@ -0,0 +1,281 @@
+    public void reportsInnerEventsAsOffending() {
+        CheckerResult result = runNemesisResilienceChecker(
+                NEMESIS_START,
+                NEMESIS_START_2,
+                NEMESIS_STOP,
+                NEMESIS_STOP_2);
+
+        assertThat(result.valid()).isFalse();
+        assertThat(result.errors()).containsExactly(NEMESIS_START_2, NEMESIS_STOP);
+    }
+
+    @Test
+    public void succeedsIfCycleBetweenInnermostEvents() {
+        assertNoErrors(NEMESIS_START, NEMESIS_START_2, INVOKE_0, OK_0, NEMESIS_STOP, NEMESIS_STOP_2);
+    }
+
+    @Test
+    public void failsIfCycleNotBetweenInnermostEvents() {
+        CheckerResult result = runNemesisResilienceChecker(
+                NEMESIS_START,
+                INVOKE_0,
+                NEMESIS_START_2,
+                OK_0,
+                INVOKE_1,
+                NEMESIS_STOP,
+                OK_1,
+                NEMESIS_STOP_2);
+
+        assertThat(result.valid()).isFalse();
+        assertThat(result.errors()).containsExactly(NEMESIS_START_2, NEMESIS_STOP);
+    }
+
+    @Test
+    public void reportsMultipleOffendingNemesisEvents() {
+        CheckerResult result = runNemesisResilienceChecker(
+                NEMESIS_START,
+                NEMESIS_STOP,
+                NEMESIS_START_2,
+                NEMESIS_STOP_2);
+
+        assertThat(result.valid()).isFalse();
+        assertThat(result.errors()).containsExactly(NEMESIS_START, NEMESIS_STOP, NEMESIS_START_2, NEMESIS_STOP_2);
+    }
+
+    @Test
+    public void onlyReportsRelevantOffendingEvents() {
+        CheckerResult result = runNemesisResilienceChecker(
+                NEMESIS_START,
+                INVOKE_0,
+                OK_0,
+                NEMESIS_STOP,
+                NEMESIS_START_2,
+                NEMESIS_STOP_2);
+
+        assertThat(result.valid()).isFalse();
+        assertThat(result.errors()).containsExactly(NEMESIS_START_2, NEMESIS_STOP_2);
+    }
+
+    @Test
+    public void failsOnDistributedCycle() {
+        CheckerResult result = runNemesisResilienceChecker(
+                NEMESIS_START,
+                INVOKE_0,
+                NEMESIS_STOP,
+                NEMESIS_START_2,
+                OK_0,
+                NEMESIS_STOP_2);
+
+        assertThat(result.valid()).isFalse();
+        assertThat(result.errors()).containsExactly(NEMESIS_START, NEMESIS_STOP, NEMESIS_START_2, NEMESIS_STOP_2);
+    }
+
+    @Test
+    public void ignoresNonNemesisRelatedInfoEvents() {
+        assertNoErrors(IMPOSTOR_START, IMPOSTOR_STOP);
+    }
+
+    @Test
+    public void ignoresNonNemesisRelatedInfoEventsMidstream() {
+        assertNoErrors(NEMESIS_START, INVOKE_0, IMPOSTOR_START, IMPOSTOR_STOP, OK_0, NEMESIS_STOP);
+    }
+
+    @Test
+    public void skipsNonNemesisRelatedInfoEvents() {
+        assertSimpleNemesisError(INVOKE_0, NEMESIS_START, IMPOSTOR_START, OK_0, NEMESIS_STOP, IMPOSTOR_STOP);
+    }
+
+    private static void assertNoErrors(Event... events) {
```

Comment:
I wonder if we could share some of these functions, since we have the same (or very similar) things in all of the other checker tests too